dev/tasks/r/github.macos.autobrew.yml

```diff
@@ -35,14 +35,20 @@ jobs:
             - "{{ macros.r_oldrel.ver }}"
     steps:
       {{ macros.github_checkout_arrow()|indent }}
+      - name: Setup hosted
+        if: matrix.platform != 'macos-10.13'
+        run: |
+          # minio and sccache are pre-installed on the self-hosted 10.13 runner.
+          # Install minio for tests.
+          brew install minio
+          brew install sccache
+
+          # Ensure removing OpenSSL from the default paths to avoid
+          # mixing OpenSSL in the default paths and OpenSSL installed
+          # by autobrew.
+          brew unlink openssl || :
```

Comment:
Hmm... can we instead make sure our build chain selects OpenSSL consistently? Users will otherwise probably encounter a similar problem if they try compiling Arrow themselves.

Reply:
That would also be my preference though it seems to be tricky and I haven't had the time to investigate.

Reply:
https://github.com/ursacomputing/crossbow/actions/runs/5504413947/jobs/10030597584#step:8:19

Reply:
I think that it's difficult.

For example, please consider the following case:
* OpenSSL 1.1 is installed in `/usr/local`
* OpenSSL 3 is install in `/opt/openssl`
* c-ares is installed in `/usr/local`
* Build bundled gRPC with c-ares and OpenSSL 3
* gRPC uses `-I/usr/local/include` from c-ares and `-I/opt/openssl/include` from OpenSSL 3
* `-I/opt/openssl/include -I/usr/local/include` will work but `-I/usr/local/include -I/opt/openssl/include` will not work because OpenSSl 1.1 headers in `/usr/local/include` are used
* I don't think that we can always use include flags from OpenSSL 3 as the first include flag because there are more version mixable libraries such as Abseil